src/translate.py

```diff
@@ -3689,17 +3805,19 @@ def translate_graph_from_block(
     for child in node.immediately_dominates:
         if isinstance(child, TerminalNode):
             continue
-        new_contents = regs.contents.copy()
+        new_contents = {
+            reg: RegData(data.value, RegDataMeta(inherited=True))
```

Comment:
Do we need to throw away the previous values in `RegDataMeta` here? It looks like `function_return=True` could be set.

Reply:
It doesn't matter either way -- for `inherited=True` metas we recompute `function_return` based on parent nodes.

(fwiw I had an earlier patch set which reused the entire old RegData here, in the case where it doesn't create a phi for the register, but it ended up less uniform and harder to understand: it made phi vs non-phi parents into separate cases for no real reason.)